test/libsolidity/semanticTests/storageLayoutSpecifier/variable_cleanup_sstore.sol

```diff
@@ -0,0 +1,16 @@
+contract C {
+    uint8 x;
+    int8 y;
+    bytes2 b;
+    // Note the return types are larger than the state variable ones
+    function f(uint _x, int _y, bytes3 _b) public returns (uint16, int32, bytes32) {
+        assembly {
+            sstore(x.slot, _x)
+            sstore(y.slot, _y)
+            sstore(b.slot, _b)
```

Comment:
This is actually wrong. Note that all variables are packed into a single slot so `b` overwrites all of them here. You need to combine them and store once if you want this to be the equivalent of Solidity version.

Reply:
Ah, right, that flew over my head.
Not sure what should be a good way to do that here though.

Reply:
You could just use types large enough that they will land in separate slots. Or use single-element static arrays.

Reply:
I wanted to make it very clear, but didn't realize that the result itself already shows exactly what is happening.